program_transformers/src/mpl_core_program/v1_asset.rs

```diff
@@ -536,3 +542,58 @@ fn convert_keys_to_snake_case(plugins_json: &mut Value) {
         _ => {}
     }
 }
+
+/// Updates the `asset_authority` for all assets that are part of a collection in a batch.
+/// This function performs a cursor-based paginated read and batch update.
+async fn update_group_asset_authorities<T: ConnectionTrait + TransactionTrait>(
+    conn: &T,
+    group_key: Vec<u8>,
+    authority: Vec<u8>,
+    slot: i64,
+) -> ProgramTransformerResult<()> {
+    let mut after = None;
+
+    let mut query = asset_grouping::Entity::find()
+        .select_only()
+        .column(asset_grouping::Column::AssetId)
+        .filter(asset_grouping::Column::GroupKey.eq(group_key.clone()))
+        .cursor_by(asset_grouping::Column::AssetId);
+    let mut query = query.first(1_000);
+
+    loop {
+        if let Some(after) = after.clone() {
+            query = query.after(after);
+        }
+
+        let entries = query.all(conn).await?;
+
+        if entries.is_empty() {
+            break;
+        }
+
+        let asset_ids = entries
+            .clone()
+            .into_iter()
+            .map(|entry| entry.asset_id)
+            .collect::<Vec<_>>();
+
+        asset_authority::Entity::update_many()
+            .col_expr(
+                asset_authority::Column::Authority,
+                Expr::value(authority.clone()),
+            )
+            .col_expr(asset_authority::Column::SlotUpdated, Expr::value(slot))
+            .filter(asset_authority::Column::AssetId.is_in(asset_ids))
+            .filter(Expr::cust_with_values(
+                "asset_authority.slot_updated <= ?",
+                vec![slot],
+            ))
```

Comment:
For some reason it doesn't like this, even though it looks like valid syntax when I ran it it never filled in the `?` with the slot value and caused a syntax error.